astroquery/linelists/cdms/tests/test_cdms.py

```diff
@@ -57,9 +61,54 @@ def test_query():
     tbl = CDMS._parse_result(response)
     assert isinstance(tbl, Table)
     assert len(tbl) == 8
-    assert set(tbl.keys()) == set(['FREQ', 'ERR', 'LGINT', 'DR', 'ELO', 'GUP',
-                                   'TAG', 'QNFMT', 'Ju', 'Jl', "vu", "vl", "Ku", "Kl", "F", "name"])
+    assert set(tbl.keys()) == colname_set
 
     assert tbl['FREQ'][0] == 115271.2018
     assert tbl['ERR'][0] == .0005
     assert tbl['LGINT'][0] == -7.1425
+
+
+def test_parseletternumber():
+    """
+    Very Important:
+    Exactly two characters are available for each quantum number. Therefore, half
+    integer quanta are rounded up ! In addition, capital letters are used to
+    indicate quantum numbers larger than 99. E. g. A0 is 100, Z9 is 359. Small
+    types are used to signal corresponding negative quantum numbers.
+    """
+
+    # examples from the docs
+    assert parse_letternumber("A0") == 100
+    assert parse_letternumber("Z9") == 359
+
+    # inferred?
+    assert parse_letternumber("z9") == -359
+    assert parse_letternumber("ZZ") == 3535
+
+
+def test_hc7s():
```

Comment:
This test fails when run on it's own, it relies on other tests prior to it run successfully, a side effect of calling the private method directly.

```
astroquery/linelists/cdms/tests/test_cdms.py F                                                                   [100%]

======================================================= FAILURES =======================================================
______________________________________________________ test_hc7s _______________________________________________________

    def test_hc7s():
        """
        Test for a very complicated molecule
    
        CDMS.query_lines_async(100*u.GHz, 100.755608*u.GHz, molecule='HC7S', parse_name_locally=True)
        """
    
        response = MockResponseSpec('HC7S.data')
>       tbl = CDMS._parse_result(response)

astroquery/linelists/cdms/tests/test_cdms.py:97: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <astroquery.linelists.cdms.core.CDMSClass object at 0x135f68df0>
response = <astroquery.linelists.cdms.tests.test_cdms.MockResponseSpec object at 0x14d075ee0>, verbose = False

    def _parse_result(self, response, verbose=False):
        """
        Parse a response into an `~astropy.table.Table`
    
        The catalog data files are composed of fixed-width card images, with
        one card image per spectral line.  The format of each card image is
        similar to the JPL version:
        FREQ, ERR, LGINT, DR,  ELO, GUP, TAG, QNFMT,  QN',  QN"
        (F13.4,F8.4, F8.4,  I2,F10.4,  I3,  I7,    I4,  6I2,  6I2)
        but the formats are somewhat different and are encoded below.
        The first several entries are the same, but more detail is appended at
        the end of the line
    
        FREQ:  Frequency of the line in MHz.
        ERR:   Estimated or experimental error of FREQ in MHz.
        LGINT: Base 10 logarithm of the integrated intensity in units of nm^2 MHz at
            300 K.
    
        DR:    Degrees of freedom in the rotational partition function (0 for atoms,
            2 for linear molecules, and 3 for nonlinear molecules).
    
        ELO:   Lower state energy in cm^{-1} relative to the ground state.
        GUP:   Upper state degeneracy.
        MOLWT: The first half of the molecular weight tag, which is the mass in atomic
               mass units (Daltons).
        TAG:   Species tag or molecular identifier.  This only includes the
               last 3 digits of the CDMS tag
    
        A negative value of MOLWT flags that the line frequency has been
        measured in the laboratory.  We record this boolean in the 'Lab'
        column.  ERR is the reported experimental error.
    
        QNFMT: Identifies the format of the quantum numbers
        Ju/Ku/vu and Jl/Kl/vl are the upper/lower QNs
        F: the hyperfine lines
        name: molecule name
    
        The full detailed description is here:
        https://cdms.astro.uni-koeln.de/classic/predictions/description.html#description
        """
    
        if 'Zero lines were found' in response.text:
            raise EmptyResponseError(f"Response was empty; message was '{response.text}'.")
    
        soup = BeautifulSoup(response.text, 'html.parser')
        text = soup.find('pre').text
    
        starts = {'FREQ': 0,
                  'ERR': 14,
                  'LGINT': 25,
                  'DR': 36,
                  'ELO': 38,
                  'GUP': 48,
                  'MOLWT': 51,
                  'TAG': 54,
                  'QNFMT': 58,
                  'Ju': 61,
                  'Ku': 63,
                  'vu': 65,
                  'F1u': 67,
                  'F2u': 69,
                  'F3u': 71,
                  'Jl': 73,
                  'Kl': 75,
                  'vl': 77,
                  'F1l': 79,
                  'F2l': 81,
                  'F3l': 83,
                  'name': 89}
    
        result = ascii.read(text, header_start=None, data_start=0,
                            comment=r'THIS|^\s{12,14}\d{4,6}.*',
                            names=list(starts.keys()),
                            col_starts=list(starts.values()),
                            format='fixed_width', fast_reader=False)
    
        result['FREQ'].unit = u.MHz
        result['ERR'].unit = u.MHz
    
        result['Lab'] = result['MOLWT'] < 0
        result['MOLWT'] = np.abs(result['MOLWT'])
        result['MOLWT'].unit = u.Da
    
        for suf in 'ul':
            for qn in ('J', 'v', 'K', 'F1', 'F2', 'F3'):
                qnind = qn+suf
                if result[qnind].dtype != int:
                    intcol = np.array(list(map(parse_letternumber, result[qnind])),
                                      dtype=int)
                    result[qnind] = intcol
    
        # if there is a crash at this step, something went wrong with the query
        # and the _last_query_temperature was not set.  This shouldn't ever
        # happen, but, well, I anticipate it will.
>       if self._last_query_temperature == 0:
E       AttributeError: 'CDMSClass' object has no attribute '_last_query_temperature'

astroquery/linelists/cdms/core.py:269: AttributeError
=============================================== short test summary info ================================================
FAILED astroquery/linelists/cdms/tests/test_cdms.py::test_hc7s - AttributeError: 'CDMSClass' object has no attribute ...
================================================== 1 failed in 0.93s ===================================================
```